fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexingHeartbeatTest.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * OnlineIndexingHeartbeatTest.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2025 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb;
+
+import com.apple.foundationdb.record.IndexBuildProto;
+import com.apple.foundationdb.record.RecordCoreException;
+import com.apple.foundationdb.record.metadata.Index;
+import com.apple.foundationdb.record.metadata.IndexOptions;
+import com.apple.foundationdb.record.metadata.IndexTypes;
+import com.apple.foundationdb.record.metadata.expressions.EmptyKeyExpression;
+import com.apple.foundationdb.tuple.Tuple;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static com.apple.foundationdb.record.metadata.Key.Expressions.field;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Verify indexing heartbeat activity (query & clear).
+ */
+class OnlineIndexingHeartbeatTest extends OnlineIndexerTest {
+
+    @Test
+    void testHeartbeatLowLevel() {
```

Comment:
It would be good to look at `SynchronizedSessionTest` and make sure you are covering the same features, probably at a low-level and at a high level.
At first glance, beyond what I've already suggested, I see that it could benefit from testing that two different indexes can be built simultaneously regardless of indexer type.

Reply:
After the recent changes, most of `SynchronizedSessionTest`'s tests are covered. I had added 
`OnlineIndexingHeartbeatTest.testHeartbeatsRenewal` which asserts that during online indexing the heartbeats are: 
* single item
* same indexerId, info, and genesis time
* monotonically increasing heartbeats